File: Python/백준/Bronze/5086. 배수와 약수/배수와 약수.py
import sys

input = sys.stdin.readline
a,b = map(int,input().split())

while a and b:
    i = 1
    if a > b:
        while 1:
            if b*i > a:
                print("neither")
                break
            
            if b*i == a:
                print("multiple")
                break

            i+=1    
    else:
        while 1:
            if b < a*i:
                print("neither")
                break
            
            if b == a*i:
                print("factor")
                break

            i+=1    
    
    
    a,b = map(int,input().split())

File: C++17/백준/Bronze/5086. 배수와 약수/배수와 약수.cc
#include <iostream>

int main(void){
    int A,B;
    
    std::cin >> A >> B;
    
    while (A+B){
        
        if (A%B == 0){
            std::cout << "multiple" << std::endl;
            std::cin >> A >> B;
            continue;
        }
        
        if (B%A == 0){
            std::cout << "factor" << std::endl;
            std::cin >> A >> B;
            continue;
        }
        std::cout << "neither" << std::endl;
        std::cin >> A >> B;
    }
    
    return 0;
}